### PR TITLE
[Core][StructuralMechanicsApplication] Improving Line2D3 and correcting load for Line2D3

### LINKED ISSUE #4272: [Structural] Convergence problem in simple case
Dear @KratosMultiphysics/structural-mechanics ,

While debugging my new embedded solver I realised that the structure was not converging for a simple `POSITIVE_FACE_PRESSURE`distribution. I did extract such pressure load to set a simple structural mechanics problem with the same geometry. Some of you will note that it is the Mok benchmark beam but with a larger Young modulus. 

I'm attaching the files so you can run it. 

Many thanks in advance!
[test_structure.zip](https://github.com/KratosMultiphysics/Kratos/files/2906492/test_structure.zip)

### PR DESCRIPTION
**Pull request by loumalouomega**
This fixes #4272. Line2D3 had several bad definitions, this fixes the Line2D3

### CODE REVIEW

**Comment by loumalouomega** [comment]
Includes #4599

**Comment by loumalouomega** [comment]
Ping @philbucher

**Comment by philbucher** [comment]
Can you merge master please?

**Comment by loumalouomega** [comment]
> Can you merge master please?

Done

File: kratos/geometries/line_2d_3.h [hunk 1]
```diff
@@ -373,21 +378,89 @@ class Line2D3 : public Geometry<TPointType>
...
-        const double Tolerance = std::numeric_limits<double>::epsilon() 
+        const double Tolerance = std::numeric_limits<double>::epsilon()
         ) override
     {
-        this->PointLocalCoordinates( rResult, rPoint );
+        PointLocalCoordinates( rResult, rPoint );
 
-        if ( std::abs( rResult[0] ) <= (1.0 + Tolerance) )
-        {
+        if ( std::abs( rResult[0] ) <= (1.0 + Tolerance) ) {
```
Comment (philbucher) [inline kratos/geometries/line_2d_3.h:388]: this check does not seem to be correct

File: kratos/geometries/line_2d_3.h [hunk 2]
```diff
@@ -373,21 +378,89 @@ class Line2D3 : public Geometry<TPointType>
...
+        Matrix J = ZeroMatrix( 1, 1 );
+        Matrix invJ = ZeroMatrix( 1, 1 );
+
+        // Starting with xi = 0
+        if (rResult.size() != 3)
+            rResult.resize(3, false);
+        rResult = ZeroVector( 3 );
+        double delta_xi = 0.0;
+        const array_1d<double, 3> zero_array = ZeroVector(3);
+        array_1d<double, 3> CurrentGlobalCoords;
```
Comment (philbucher) [inline kratos/geometries/line_2d_3.h]:
```suggestion
        array_1d<double, 3> current_global_coords;
```

File: kratos/geometries/line_2d_3.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/geometries/line_2d_3.h:388]: It is correct, the number of local coordinates is 1, between -1 and 1, it is exactly the same as the line with two nodes

Comment (loumalouomega) [reply kratos/geometries/line_2d_3.h:388]: https://github.com/KratosMultiphysics/Kratos/blob/310b2f21ac91bbeb71aef35bdaad8c6a229d7850/kratos/geometries/line_2d_2.h#L888

Comment (philbucher) [reply kratos/geometries/line_2d_3.h:388]: sorry, my bad, now I understood, nevermind

**Review by philbucher — APPROVED** [review]
@KratosMultiphysics/technical-committee I am approving this since I think the changes are correct, it seems to be working fine with the lineloadcondition in StructMech
If someone has some comments then please let use know
Otherwise those PRs usually hang around forever ...

Thanks for cleaning @loumalouomega

**Pull request merged**